File: app/src/main/java/me/devsaki/hentoid/activities/sources/MusesActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class MusesActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "8muses.com";
    private static final String[] GALLERY_FILTER = {"//www.8muses.com/comics/album/", "//comics.8muses.com/comics/album/"};
//    private static final String[] DIRTY_ELEMENTS = {".c-tile:not([href])"}; // <-- even when removing empty tiles, ads are generated and force-inserted by the ad JS (!)

    Site getStartSite() {
        return Site.MUSES;
    }

    @Override
    protected CustomWebViewClient getWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
//        client.addDirtyElements(DIRTY_ELEMENTS);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/MusesActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

class MusesActivity : BaseWebActivity() {

    companion object {
        private const val DOMAIN_FILTER = "8muses.com"
        private val GALLERY_FILTER =
            arrayOf("//www.8muses.com/comics/album/", "//comics.8muses.com/comics/album/")
        // private static final String[] REMOVABLE_ELEMENTS = {".c-tile:not([href])"}; // <-- even when removing empty tiles, ads are generated and force-inserted by the ad JS (!)
    }

    override fun getStartSite(): Site {
        return Site.MUSES
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.adBlocker.addToJsUrlWhitelist(DOMAIN_FILTER)
        return client
    }
}